app/schemas/selectors/cinematic.js

```diff
@@ -0,0 +1,132 @@
+/**
+ * Selector / verifier.
+ *
+ * Querying a nested data structure gets very repetitive.
+ * This file contains selectors for various elements in the cinematic schema.
+ *
+ * Certain selectors can also throw an exception if correct data can't be returned.
+ * This is useful when certain properties need to be returned together.
+ * Otherwise if a selector can't be fulfilled we should return undefined.
+ *
+ * For example we may have an enum string that should return different properties
+ * depending on how that enum has been set. This should throw an error if the shape
+ * of the data is invalid.
+ *
+ * ## Examples
+ *
+ * If we ask for the left character and it's not there, return undefined.
+ * If it's there and the enum type has been set to `slug`. This may indicate
+ * that we return an object with a `slug`. Or maybe the enum type is `hero` and
+ * no additional properties are required. In this case if the type is `slug` and no
+ * slug is present, an exception should be thrown.
+ *
+ */
+
+/**
+ * Composes a list of functions.
+ * The initial argument is passed into the first function and the result
+ * is the passed along the array. The final result is returned.
+ * @param  {...any} fns List of functions
+ */
+const compose = (...fns) => initial => fns.reduce((v, fn) => fn(v), initial)
+
+/**
+ * @typedef {Object} CharacterSchema
+ * @property {string} type
+ * @property {string} slug
+ * @property {boolean} enterOnStart
+ * @property {Point2d} position
+ */
+
+/**
+ * @typedef {Object} ShotSetup - Shot setup object
+ * @property {string} cameraType - The camera type enum
+ * @property {CharacterSchema} rightThangType
+ * @property {CharacterSchema} leftThangType
+ */
+
+/**
+ * @typedef {Object} Point2d
+ * @property {number} x
+ * @property {number} y
+ */
+
+/**
+ * @typedef {Object} DialogNode
+ * @property {string} text - The text to display
+ * @property {Point2d} textLocation - The point information
```

Comment:
I see that these do not include all the properties that are in the schema, is that because we only include the ones that we are using/returning in this file?

Reply:
Yes. We can extend this as we add more validators and selectors.

These are also the only ones tested.  (I didn't want to add things that may change as features are added).